sklearn/metrics/tests/test_pairwise.py

```diff
@@ -331,6 +333,47 @@ def test_pairwise_parallel(func, metric, kwds, dtype):
     assert_allclose(S, S2)
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize(
+    "func, metric, kwds",
+    [
+        (pairwise_distances, "euclidean", {}),
+        (
+            pairwise_distances,
+            minkowski,
+            _minkowski_kwds,
+        ),
+        (
+            pairwise_distances,
+            "minkowski",
+            _minkowski_kwds,
+        ),
```

Comment:
Since scipy's `cdist` is not array API aware yet and we have no visibility on when it will happen if it does, it's quite unlikely to be able to support array API for all metrics names. Instead we should pickle a small subset of distance metrics, e.g. those from `PAIRWISE_DISTANCE_FUNCTIONS` that already have array API support and document explicit in the `doc/modules/array_api.rst` page which subset we currently support.

Then we need to align our array API test parametrization to cover consistently this subset.